pkg/docker/util_test.go

```diff
@@ -117,14 +119,43 @@ func TestCheckAllowedUser(t *testing.T) {
 			onbuild:     []string{"RUN echo \"hello world\"", "USER 10:wheel"},
 			expectErr:   false,
 		},
+		{
+			name:         "AllowedUIDs is set, numeric user, assemble user override ok",
+			allowedUIDs:  rangeList("1-"),
+			user:         "200",
+			assembleUser: "10",
+			expectErr:    false,
+		},
+		{
+			name:         "AllowedUIDs is set, numeric user, root assemble user",
+			allowedUIDs:  rangeList("1-"),
+			user:         "200",
+			assembleUser: "0",
+			expectErr:    true,
+		},
+		{
+			name:        "AllowedUIDs is set, numeric user, assemble user label ok",
+			allowedUIDs: rangeList("1-"),
+			user:        "200",
+			labels:      map[string]string{AssembleUserLabel: "10"},
+			expectErr:   false,
+		},
+		{
+			name:        "AllowedUIDs is set, numeric user, assemble user label root",
+			allowedUIDs: rangeList("1-"),
+			user:        "200",
+			labels:      map[string]string{AssembleUserLabel: "0"},
+			expectErr:   true,
+		},
```

Comment:
need a case where the assembleuser is ok but the imageuser is not ok (and it should be allowed)